src/sql/base/browser/ui/taskbar/actionbar.ts

```diff
@@ -128,15 +141,144 @@ export class ActionBar extends ActionRunner implements IActionRunner {
+		let item = this._actionsList.removeChild(this._actionsList.childNodes[index]);
+		this._overflow.insertBefore(item, this._overflow.firstChild);
+
+		// change role to menuItem when it's in the overflow
+		if ((<HTMLElement>this._overflow.firstChild).className !== 'taskbarSeparator') {
+			(<HTMLElement>this._overflow.firstChild.firstChild).setAttribute('role', 'menuItem');
+		}
+	}
+
+	private createMoreItemElement(): void {
+		this._moreItemElement = document.createElement('li');
+		this._moreItemElement.className = 'action-item more';
+		this._moreItemElement.setAttribute('role', 'presentation');
+		this._moreActionsElement = document.createElement('a');
+		this._moreActionsElement.className = 'moreActionsElement action-label codicon toggle-more';
+		this._moreActionsElement.setAttribute('role', 'button');
+		this._moreActionsElement.title = nls.localize('toggleMore', "Toggle More");
+		this._moreActionsElement.tabIndex = 0;
+		this._moreActionsElement.setAttribute('aria-haspopup', 'true');
+		this._register(DOM.addDisposableListener(this._moreActionsElement, DOM.EventType.CLICK, (e => { this.toggleOverflowDisplay(); })));
+		this._register(DOM.addDisposableListener(this._moreActionsElement, DOM.EventType.KEY_DOWN, (ev => {
+			let event = new StandardKeyboardEvent(ev);
+			if (event.keyCode === KeyCode.Enter || event.keyCode === KeyCode.Space) {
+				this._focusedItem = undefined; // so that the default actionbar click handler doesn't trigger the selected action-item
+				this.toggleOverflowDisplay();
+				DOM.EventHelper.stop(event, true);
+			}
+		})));
+
+		this._register(DOM.addDisposableListener(this._overflow, DOM.EventType.KEY_DOWN, (e: KeyboardEvent) => {
+			let event = new StandardKeyboardEvent(e);
+
+			// Close overflow if Escape is pressed
+			if (event.equals(KeyCode.Escape)) {
+				this.hideOverflowDisplay();
+				this._moreActionsElement.focus();
+			} else if (event.equals(KeyCode.UpArrow) && this._focusedItem !== this._actionsList.childNodes.length - 1) { // up arrow on first element in overflow should not move out from the overflow
+				this.focusPrevious();
+			} else if (event.equals(KeyCode.DownArrow) && this._focusedItem !== this._actionsList.childNodes.length + this._overflow.childNodes.length - 2) { // down arrow on last element shouldn't move out from overflow
+				this.focusNext();
+			} else if (event.equals(KeyMod.Shift | KeyCode.Tab)) {
+				this._moreActionsElement.focus();
+			} else if (event.equals(KeyCode.Tab)) {
+				this.hideOverflowDisplay();
+				(<HTMLElement>this._actionsList.parentElement.nextElementSibling).focus();
+			}
+			DOM.EventHelper.stop(event, true);
+		}));
+
+		this._moreItemElement.appendChild(this._moreActionsElement);
+		this._actionsList.appendChild(this._moreItemElement);
+		let placeHolderItem = new ActionViewItem(this, undefined, undefined);
+		placeHolderItem.actionRunner = undefined;
+		this._items.push(undefined); // add place holder for more item element
```

Comment:
Did you mean to push placeHolderItem here? It's not doing anything otherwise...

Reply:
oops forgot to remove this. Ended up just using undefined as the placeholder instead of placeHolderItem